### PR TITLE
Beginning of adding a Cyborg job slot to Talon

### PR DESCRIPTION
**Pull request by Rykka-Stormheart**
Adds a Cyborg job slot to Talon crew, same 14 day limit as the rest of the crew, and adds a spawn/leave storage spot as well as a recharger next to it.

Modules are Tether default, access is Tether default, I'll have to figure out how to change that, because a lot of the Cyborg/AI stuff is hardcoded to assume there's only ever one main "station", it seems like?

I'm very new to this, so I'm probably doing it wrong.

### CODE REVIEW

File: maps/tether/submaps/offmap/talon1.dmm [hunk 1]
```diff
@@ -7281,14 +7302,6 @@
 /obj/structure/disposalpipe/segment,
 /turf/simulated/floor/tiled/steel,
 /area/talon/deckone/central_hallway)
-"PY" = (
-/obj/machinery/firealarm{
```
Comment (Rykka-Stormheart) [inline maps/tether/submaps/offmap/talon1.dmm:7285]: I'll fix this in a bit, in case anyone's worried about the missing fire alarm.

**Comment by Arokha** [comment]
Silicons have an `idcard_type` var that determines what sort of internal ID card they spawn inside them that has all their access on it. Set theirs to that talon ID I made.

**Comment by Rykka-Stormheart** [comment]
> Silicons have an `idcard_type` var that determines what sort of internal ID card they spawn inside them that has all their access on it. Set theirs to that talon ID I made.

What about the modules? Certain modules give access to channels like Service, etc etc.

**Comment by Arokha** [comment]
All borgs have all access. Do you mean for radio stuff?

**Comment by Rykka-Stormheart** [comment]
> All borgs have all access. Do you mean for radio stuff?

Ye. The "Service" Dogborg module gives access to the Tether's "Service" channel, same with the rest of the dogborg modules. Not sure how to correct that myself.

**Comment by Arokha** [comment]
Not sure either. I can look tomorrow mebbe.

**Comment by Arokha** [comment]
Ok, found it. Here's an example:
```
/obj/item/weapon/robot_module/robot/security
	name = "security robot module"
	channels = list("Security" = 1)
	networks = list(NETWORK_SECURITY)
```

So for a talon one, you'd want to do
```
/obj/item/weapon/robot_module/robot/talon
	name = "talon robot module"
	channels = list("Talon" = 1)
	networks = list(NETWORK_TALON_HELMETS, NETWORK_TALON_SHIP)
```

**Comment by Rykka-Stormheart** [comment]
> Ok, found it. Here's an example:
> 
> ```
> /obj/item/weapon/robot_module/robot/security
> 	name = "security robot module"
> 	channels = list("Security" = 1)
> 	networks = list(NETWORK_SECURITY)
> ```
> 
> So for a talon one, you'd want to do
> 
> ```
> /obj/item/weapon/robot_module/robot/talon
> 	name = "talon robot module"
> 	channels = list("Talon" = 1)
> 	networks = list(NETWORK_TALON_HELMETS, NETWORK_TALON_SHIP)
> ```

I'll get on it once I'm more awake, thanks! <3

File: maps/tether/submaps/offmap/talon1.dmm (hunk 1 above)
Comment (Rykka-Stormheart) [reply maps/tether/submaps/offmap/talon1.dmm:7296]: Fixed, resolving this, fire alarm is still in the robotics room

**Comment by Rykka-Stormheart** [comment]
So, as it turns out, there's so many things bound up in robot.dm and ai stuff that makes calls to each other, and the fact that both AI and robots/cyborgs are coded to assume there's only ever one AI, and only ever one faction for Tether, I'm closing this and I'm going to see if I can modularize the robot/AI code.

For instance:
`	var/changed_name = ""
	if(custom_name)
		changed_name = custom_name
		notify_ai(ROBOT_NOTIFICATION_NEW_NAME, real_name, changed_name)`
This calls to tell the AI that there was a name change. It assumes there's only ever one AI.

`idcard_type will not work for talon borgs
Rykka Stormheart Today at 2:22 AM
because borgs are hardcoded to have all access, iirc?
Shadow Larkens Today at 2:22 AM
nah, because idcard_type is used during robot/New()
which means that as soon as Robotize() is called, the id card will be created`

So yeah, dog with very little knowledge of things is going to try a refactor with some help... or the project might get dropped if it's too far out of my knowledge.

Sorry to those that wanted/were interested in a Talon borg.
Wish me luck. ;~;

**Comment by Heroman3003** [comment]
If you want to look for things potentially untied to the station, I suggest looking into Syndie borg modules.

**Pull request closed without merging**